rules/aip0203/field_behavior_required_test.go

```diff
@@ -131,12 +201,90 @@ func TestFieldBehaviorRequiredNested(t *testing.T) {
 					// of field behavior in the response.
 					string name = 1;
 				}
-			`, test)
+			`, tc)
+
 			it := f.GetServices()[0].GetMethods()[0].GetInputType()
 			nestedField := it.GetFields()[0].GetMessageType().GetFields()[0]
-			if diff := test.problems.SetDescriptor(nestedField).Diff(fieldBehaviorRequired.Lint(f)); diff != "" {
+
+			if diff := tc.problems.SetDescriptor(nestedField).Diff(fieldBehaviorRequired.Lint(f)); diff != "" {
 				t.Errorf(diff)
 			}
 		})
 	}
 }
+
+func TestFieldBehaviorRequired_NestedMessages_MultipleFile(t *testing.T) {
+	testCases := []struct {
+		name             string
+		MessageType      string
+		MessageFieldName string
+		problems         testutils.Problems
+	}{
+		{
+			"ValidAnnotatedAndChildAnnotated",
+			"Annotated",
+			"annotated",
+			nil,
+		},
+		{
+			"InvalidChildNotAnnotated",
+			"NonAnnotated",
+			"non_annotated",
+			testutils.Problems{{Message: "must be set"}},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			f1 := `
+				import "google/api/field_behavior.proto";
+				import "resource.proto";
+
+				service Library {
+					rpc UpdateBook(UpdateBookRequest) returns (UpdateBookResponse) {
+					}
+				}
+
+				message UpdateBookRequest {
+					{{.MessageType}} {{.MessageFieldName}} = 1 [(google.api.field_behavior) = REQUIRED];
+				}
+
+				message UpdateBookResponse {
+					// verifies that no error was raised on lack
+					// of field behavior in the response.
+					string name = 1;
+				}
+			`
+
+			f2 := `
+				import "google/api/field_behavior.proto";
+
+				message NonAnnotated {
+					string nested = 1;
+				}
+
+				message Annotated {
+					string nested = 1 [(google.api.field_behavior) = REQUIRED];
+				}
+			`
+
+			srcs := map[string]string{
+				"service.proto":  f1,
+				"resource.proto": f2,
+			}
+
+			ds := testutils.ParseProto3Tmpls(t, srcs, tc)
+			f := ds["service.proto"]
+			it := f.GetServices()[0].GetMethods()[0].GetInputType()
+			fd := it.GetFields()[0].GetMessageType().GetFields()[0]
+
+			if diff := tc.problems.SetDescriptor(fd).Diff(fieldBehaviorRequired.Lint(f)); diff != "" {
+				t.Errorf(diff)
+			}
+
+			want := "resource.proto"
+			if got := fd.GetFile().GetName(); got != want {
```

Comment:
Sorry, how does this actually check that the `Descriptor` included in the `Problem` is from the file not being directly linted? I think this is just testing that the template files were parsed correctly?

Reply:
The file being linted is "service.proto", the field in the problem is in "resource.proto".